### ISSUE TITLE
Adding a .txt file inside a toys folder throws a `NoMethodError`

### ISSUE THREAD

**Issue by carlo** [dazuma/toys#14]
#### Issue description

When adding a text file (non-`.rb`) to a toys folder, `toys` will throw a `NoMethodError`.

Background: I wanted to save a really long description as a separate file next to the actual tool, think `customers/create-location.rb` (the task) and `customers/create-location.txt`.

#### Steps to reproduce the issue

1. Create a bare-bones working task file, e.g. `example/my-task.rb`.
2. Create a sibling file, `example/my-task.txt`.
3. Run `toys example my-task`.

#### What's the expected result?

The task runs.

#### What's the actual result?

`toys` exits with a `NoMethodError`.

#### Additional details / screenshot

Stack trace:

```
NoMethodError: undefined method `source_type' for nil:NilClass
     29: /Users/whatever/.gem/ruby/2.6.3/bin/toys:23:in `<main>'
     28: /Users/whatever/.gem/ruby/2.6.3/bin/toys:23:in `load'
     27: /Users/whatever/.gem/ruby/2.6.3/gems/toys-0.8.0/bin/toys:30:in `<top (required)>'
     26: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/cli.rb:396:in `run'
     25: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/errors.rb:111:in `capture'
     24: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/cli.rb:397:in `block in run'
     23: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:152:in `lookup'
     22: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:152:in `loop'
     21: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:153:in `block in lookup'
     20: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:448:in `load_for_prefix'
     19: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:448:in `each'
     18: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:453:in `block in load_for_prefix'
     17: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:473:in `load_validated_path'
     16: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:486:in `load_relevant_path'
     15: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:486:in `each'
     14: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:487:in `block in load_relevant_path'
     13: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:506:in `load_child_in'
     12: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:473:in `load_validated_path'
     11: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:486:in `load_relevant_path'
     10: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:486:in `each'
      9: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:487:in `block in load_relevant_path'
      8: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:506:in `load_child_in'
      7: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:473:in `load_validated_path'
      6: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:486:in `load_relevant_path'
      5: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:486:in `each'
      4: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:487:in `block in load_relevant_path'
      3: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:506:in `load_child_in'
      2: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:473:in `load_validated_path'
      1: /Users/whatever/.gem/ruby/2.6.3/gems/toys-core-0.8.0/lib/toys/loader.rb:480:in `load_relevant_path'
Error finding tool definition
    NoMethodError: undefined method `source_type' for nil:NilClass
```

**Comment by dazuma** [comment]
Thanks much for your report!

You should be able to use the "data file" feature for this purpose. See [this section in the user guide](https://dazuma.github.io/toys/gems/toys/v0.9.3/file.guide.html#Data_files). (In fact, I think the user guide actually uses your use case as an example.)

That said, it's absolutely a bug for a non-`.rb` file in the directory to cause Toys to crash. I'll get that fixed.

**Comment by carlo** [comment]
Thanks, on both accounts! The data file feature was news to me — my bad.

**Comment by dazuma** [comment]
Fix released in [toys 0.9.4](https://rubygems.org/gems/toys/versions/0.9.4).
